File: CRUDBarangApp/app/src/main/java/com/example/crudbarangapp/model/Barang.java
package com.example.crudbarangapp.model;

public class Barang {

    private String _id;
    private String nama_barang;
    private int harga;
    private int stok;
    private String deskripsi;

    public String get_id() { return _id; }
    public String getNama_barang() { return nama_barang; }
    public int getHarga() { return harga; }
    public int getStok() { return stok; }
    public String getDeskripsi() { return deskripsi;}

    public Barang(String nama_barang, int harga, int stok, String deskripsi) {
        this.nama_barang = nama_barang;
        this.harga = harga;
        this.stok = stok;
        this.deskripsi = deskripsi;
    }
}

File: crud_barang_api/models/barang.js
const mongoose = require("mongoose");

const BarangSchema = new mongoose.Schema({
  nama_barang: String,
  harga: Number,
  stok: Number,
  deskripsi: String
}, {
  collection: "barang" 
});

module.exports = mongoose.model("Barang", BarangSchema);